### PR TITLE
feat(Activity): add missing fields

### PR DESCRIPTION
**Pull request by advaith1**
**Please describe the changes this PR makes and why it should be merged:**
Adds `id`, `platform`, `syncID`, `sessionID`, and `buttons` to the Activity object (https://github.com/discord/discord-api-docs/pull/2219)

wasn't sure if I should do something for ClientPresence because bots can't send these, but it seems to already have code to handle many properties that bots can't send [[msg](https://discord.com/channels/222078108977594368/682166281826598932/774483702045736990)]

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -30,11 +32,13 @@ declare module 'discord.js' {
     public details: string | null;
     public emoji: Emoji | null;
     public flags: Readonly<ActivityFlags>;
+    public id: string;
     public name: string;
     public party: {
       id: string | null;
       size: [number, number];
     } | null;
+    public platform: ActivityPlatform;
```
Comment (Vaporox) [inline typings/index.d.ts:41]:
```suggestion
    public platform: ActivityPlatform | null;
```

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]
Waiting for the linked PR to land first, but if there are no further changes, LGTM.

File: typings/index.d.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ declare module 'discord.js' {
 
   //#region Classes
 
+  type ActivityPlatform = 'xbox' | 'samsung';
```
Comment (SpaceEEC) [inline typings/index.d.ts:24]: `type`s should go into the `Typedefs` region.

Above `ActivityType` on line 2137.

**Review by SpaceEEC — DISMISSED** [review]

**Review by iCrawl — APPROVED** [review]

File: src/structures/Presence.js [hunk 3]
```diff
@@ -194,6 +208,18 @@ class Activity {
         }
       : null;
 
+    /**
+     * The ID of the song on Spotify
+     * @type {?string}
+     */
+    this.syncID = data.sync_id || null;
```
Comment (Vaporox) [inline src/structures/Presence.js:215]:
```suggestion
    this.syncID = data.sync_id ?? null;
```

File: src/structures/Presence.js [hunk 4]
```diff
@@ -194,6 +208,18 @@ class Activity {
...
+
+    /**
+     * The platform the game is being played on
+     * @type {?ActivityPlatform}
+     */
+    this.platform = data.platform || null;
```
Comment (Vaporox) [inline src/structures/Presence.js:221]:
```suggestion
    this.platform = data.platform ?? null;
```

File: src/structures/Presence.js [hunk 5]
```diff
@@ -222,6 +248,12 @@ class Activity {
      */
     this.emoji = data.emoji ? new Emoji(presence.client, data.emoji) : null;
 
+    /**
+     * The ID of the game or Spotify session
+     * @type {?string}
+     */
+    this.sessionID = data.session_id || null;
```
Comment (Vaporox) [inline src/structures/Presence.js:255]:
```suggestion
    this.sessionID = data.session_id ?? null;
```

**Comment by advaith1** [comment]
@Vaporox all the other ones use `||`, why make these different? probably won't actually make a difference here

**Comment by kyranet** [comment]
> @Vaporox all the other ones use `||`, why make these different? probably won't actually make a difference here

Because we landed requirement for Node.js v14 (which implements `??`) fairly recently, and we haven't refactored all the eligible `||`s.

**Comment by advaith1** [comment]
So will we use `??` as the default going forward?

**Comment by kyranet** [comment]
> So will we use `??` as the default going forward?

That's correct.

**Comment by advaith1** [comment]
oh looks like `syncID` was already there but undocumented and doesn't default to null 🤔 should I remove the old one?
https://github.com/discordjs/discord.js/pull/4984/files#diff-72d7646d708e843ab070501124748d67961184ff3497db3f56dd5e6ac1f2ed92R237

**Review by SpaceEEC — APPROVED** [review]
Good to go, once the upstream PR is merged.

**Pull request merged**